app/Queries/Feeds/TrendingHashtags.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Queries\Feeds;
+
+use App\Models\Hashtag;
+use Illuminate\Database\Eloquent\Builder;
+
+final readonly class TrendingHashtags
+{
+
+    /**
+     * The max days since posted for the trending feed.
+     */
+    private const int MAX_DAYS_SINCE_POSTED = 1;
+
+    /**
+     * Get the query builder for the feed.
+     *
+     * @return Builder<Question>
+     */
+    public function builder(): Builder
+    {
+        $maxDaysSincePosted = self::MAX_DAYS_SINCE_POSTED;
+
+        return Hashtag::query()
+            ->withCount('questions')
+            ->where('created_at', '>=', now()->subDays($maxDaysSincePosted));
```

Comment:
If I'm thinking about this correctly, this `where()` clause won't do what you're intending.

The `created_at` column of our `hashtags` table is the datetime that the hashtag (unique by name, case sensitive) was **first** used in a post. It is not the time that it was _most recently_ used.

The `where()` clause you have here would provide the `most used hashtags that were first-used within the past day`. 

To truly get the "most used hashtags included in posts that were published in the past X days," you will need to incorporate the pivot table `hashtag_question` and question's `answer_created_at`. Something like:

```php
App\Models\Hashtag::query()
    ->withCount([
        'questions as included_in_recent_questions_count' => function (Builder $q) {
            $q->where($q->qualifyColumn('answer_created_at'), '>=', now()->subDays(...));
        },
    ])
    ->orderByDesc('included_in_recent_questions_count')
    ...
```

If we wanted to ensure we don't return hashtags that have a count of zero uses in the past X days, we could also tack on

```php
...
->withCount(...)
->whereHas('questions', function (Builder $q) {
        $q->where($q->qualifyColumn('answer_created_at'), '>=', now()->subDays(...));
    })
...
```

Unfortunately, I don't believe this will be an efficient query with our current schema and indexes but it may be fine for the current size of our data. I ran a query plan analysis and I believe this will involve a table scan.

If it becomes necessary, we can make this more efficient by either denormalizing our schema to include a datetime on the `hashtag_question` pivot table (to quickly reference when a particular hashtag was used in a post), or it may also be sufficient to index the `questions.answer_created_at` column.

P.S. When I use `answer_created_at`, it is basically synonymous with `created_at` for most cases currently used on Pinkary; however, they are technically different.